### PR TITLE
[FEATURE] Pull Request options

### LINKED ISSUE #4: [bug] somehow it detects commits that i have since force-pushed myself away from
in my repo i had a bad commit, so i renamed it and force pushed in my PR however the check still fails somehow, and shows the old message. It seems it catches something in the wrong way.

On top of this, if i let this run on my master branch, it will fail on my newest commit because i have a really old commit that doesn't conform the my naming convetion, rather than only checking the commit at hand. should i use another regex flag? (for thsi reason i disabled the check on anything but pull requests)

```yml
name: "Commit Message Check"
on:
  pull_request:
    types:
      - opened
      - edited
      - reopened
      - synchronize

jobs:
  check-commit-message:
    name: Check Commit Message
    runs-on: ubuntu-latest
    steps:
      - name: Check Commit Type
        uses: gsactions/commit-message-checker@v1
        with:
          pattern: '(ID-\d+|FIX): .+'
          flags: "gm"
          error: "Naming convention is `[ID]: [summary]` i.e `ID-404: Fixed My Shit`"

```

### LINKED ISSUE #16: [Feature Request] - Option to disable the title check of a PR
I tested this PR #12 and it is working, but I noticed that the PR's title can trigger the rule, is it possible to add an option to remove it too?

https://github.com/gomex/totalcross/runs/1005076797

### PR DESCRIPTION
**Pull request by ItaloYeltsin**
This PR is related to the issues #12 and #16. This PR make it possible to ignore PR payloads (title and body) and check all commit messages associated with the PR being checked. See the example bellow:
**obs-1:** do not squash, all commit are logically atomic. 

```yml
name: 'Commit Message Check'
on:
  pull_request:
    types:
      - opened
      - edited
      - reopened
      - synchronize
  push:
    branches:
      - master
      - 'releases/*'

jobs:
  check-commit-message:
    name: Check Commit Message
    runs-on: ubuntu-latest
    steps:
      - name: Check Commit Type
        uses: gsactions/commit-message-checker@v1
        with:
          pattern: '\[[^]]+\] .+$'
          flags: 'gm'
          error: 'Your first line has to contain a commit type like "[BUGFIX]".'
      - name: Check Line Length
        uses: gsactions/commit-message-checker@v1
        with:
          pattern: '^[^#].{74}'
          error: 'The maximum line length of 74 characters is exceeded.'
          excludeDescription: 'true' # optional: this excludes the description body of a pull request
          excludeTitle: 'true' # optional: this excludes the title of a pull request
          checkAllCommitMessages: 'true' # optional: this checks all commits associated with a pull request
          accessToken: ${{secrets.ACCESS_TOKEN}} # only required if checkAllCommitMessages is true
      - name: Check for Resolves / Fixes
        uses: gsactions/commit-message-checker@v1
        with:
          pattern: '^.+(Resolves|Fixes): \#[0-9]+$'
          error: 'You need at least one "Resolves|Fixes: #<issue number>" line.'
```

Resolves #16
Fixes #4

### CODE REVIEW

**Review by otavio — APPROVED** [review]

**Comment by ItaloYeltsin** [comment]
@gilbertsoft could you define the secret GITHUB_ACCESS_TOKEN with a github access token (no need for writing permissions) and re-run the test actions?

**Comment by gilbertsoft** [comment]
> @gilbertsoft could you define the secret GITHUB_ACCESS_TOKEN with a github access token (no need for writing permissions) and re-run the test actions?

Hey @ItaloYeltsin , thanks for your contribution! Looks like _GITHUB_ACCESS_TOKEN_ is no valid name for a secret, I'm always getting _Failed to add secret. Name is invalid_. Can you please choose an other name. Thx.

**Comment by ItaloYeltsin** [comment]
Hello @gilbertsoft, I've changed to `ACCESS_TOKEN`.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       - uses: actions/checkout@v1
       - run: |
           npm install
-          npm run all
+          GITHUB_ACCESS_TOKEN=${{ secrets.ACCESS_TOKEN }} npm run all
```
Comment (gilbertsoft) [inline .github/workflows/test.yml]: Does still not work, lets try it with a env option for this step see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsenv

**Review by gilbertsoft — CHANGES_REQUESTED** [review]
We're getting closer....

File: .github/workflows/test.yml (hunk 1 above)
Comment (gilbertsoft) [reply .github/workflows/test.yml]: Perhaps you also have to change the env var name, not sure.

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: I've just added the env option and still failed

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: https://github.com/GsActions/commit-message-checker/blob/1a845b6ecf28d47368ddeb8b41021ca416056de3/.github/workflows/test.yml#L14-L18

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: is this right?

Comment (gilbertsoft) [reply .github/workflows/test.yml]: Yeah, saw it. As said perhaps you also have to rename the env var

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: ok, I will rename it

Comment (gilbertsoft) [reply .github/workflows/test.yml]: It would try it now with `ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}`

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: ACCESS_TOKEN is still undefined in `input-test.test.js`

Comment (gilbertsoft) [reply .github/workflows/test.yml]: But I was right not using GITHUB_ see https://docs.github.com/en/actions/reference/environment-variables#naming-conventions-for-environment-variables

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: https://github.com/ItaloYeltsin/commit-message-checker/blob/b84ee10a68c2d69efd35e01c52be9cdb6b71cd61/.github/workflows/test.yml#L10-L18

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: https://github.com/ItaloYeltsin/commit-message-checker/blob/b84ee10a68c2d69efd35e01c52be9cdb6b71cd61/.github/workflows/test.yml#L10-L18

Now the action stoped without any result.

Comment (gilbertsoft) [reply .github/workflows/test.yml]: Yep, I guess the definition is wrong. Try this please:
```
      - uses: actions/checkout@v1
      - name: Test
        env: 
          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
        run: |
...
```

Comment (gilbertsoft) [reply .github/workflows/test.yml]: Or better use the name "Build & Test"

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: it didn't work =(

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]:
```yaml
      - name: Build & Test
        run: |
          ACCESS_TOKEN=${{ secrets.ACCESS_TOKEN }} npm install
          ACCESS_TOKEN=${{ secrets.ACCESS_TOKEN }} npm run all
````
There is something wrong with passing env variables to npm install, not even this is working

Comment (gilbertsoft) [reply .github/workflows/test.yml]: I've created a new secret "MYSECRET" which you can use to play around and also expose to logs etc, it simply contains "mygreatsecret". I currently have no idea what's the problem here, according the GH Actions help this should work...

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: https://stackoverflow.com/questions/57431439/how-to-set-an-environment-variable-to-be-used-during-npm-install

maybe I should try this first.

Comment (gilbertsoft) [reply .github/workflows/test.yml]: I would start doing testing like done here using echo https://docs.github.com/en/actions/reference/environment-variables#about-environment-variables. Therefor I created the new secret. But anyway the token has read access only, so you also could use the real token...

Comment (gilbertsoft) [reply .github/workflows/test.yml]: grafik

Comment (gilbertsoft) [reply .github/workflows/test.yml]: Have a look here https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

You should be able to use GITHUB_TOKEN instead, the only passed secret for PRs from forks.

Comment (gilbertsoft) [reply .github/workflows/test.yml]: @ItaloYeltsin Have a look at my last comment

Comment (gilbertsoft) [reply .github/workflows/test.yml]: And please use the env option if possible, looks nicer...

Comment (gilbertsoft) [reply .github/workflows/test.yml]: Oh no, you can just access it in the tests, no need to define something, it's already there :)

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: ok

File: __tests__/input-helper.test.ts [hunk 2]
```diff
@@ -178,27 +178,101 @@ describe('input-helper tests', () => {
...
+          name: 'totalcross'
+        }
+      }
+    }
+    inputs.pattern = 'some-pattern'
+    inputs.error = 'some-error'
+    inputs.excludeDescription = 'true'
+    inputs.excludeTitle = 'true'
+    inputs.checkAllCommitMessages = 'true'
+    inputs.accessToken = process.env.ACCESS_TOKEN
```
Comment (gilbertsoft) [inline __tests__/input-helper.test.ts:255]:
```suggestion
    inputs.accessToken = process.env.GITHUB_TOKEN
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -11,9 +11,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v1
-      - run: |
+      - name: Build & Test
+        env:
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+        run: |
```
Comment (gilbertsoft) [inline .github/workflows/test.yml]:
```suggestion
      - run: |
```

**Review by gilbertsoft — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: So, should I change it to GITHUB_TOKEN?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -14,6 +14,7 @@ jobs:
       - run: |
```
Comment (gilbertsoft) [inline .github/workflows/test.yml]:
```suggestion
      - name: Build & Test
        env:
          ACCESS_TOKEN: ${{ GITHUB_TOKEN }}
        run: |
```

File: __tests__/input-helper.test.ts [hunk 5]
```diff
@@ -178,27 +178,101 @@ describe('input-helper tests', () => {
...
+          name: 'totalcross'
+        }
+      }
+    }
+    inputs.pattern = 'some-pattern'
+    inputs.error = 'some-error'
+    inputs.excludeDescription = 'true'
+    inputs.excludeTitle = 'true'
+    inputs.checkAllCommitMessages = 'true'
+    inputs.accessToken = process.env.GITHUB_TOKEN
```
Comment (gilbertsoft) [inline __tests__/input-helper.test.ts]:
```suggestion
    inputs.accessToken = process.env.ACCESS_TOKEN
```

**Review by gilbertsoft — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -11,9 +11,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v1
-      - run: |
+      - name: Build & Test
+        env:
+          ACCESS_TOKEN: ${{ GITHUB_TOKEN }}
```
Comment (gilbertsoft) [inline .github/workflows/test.yml]:
```suggestion
          ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Review by gilbertsoft — CHANGES_REQUESTED** [review]

**Review by gilbertsoft — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (gilbertsoft) [reply .github/workflows/test.yml]: I already did the changes and we got a green build now :)

Comment (ItaloYeltsin) [reply .github/workflows/test.yml]: Nice, it seems it worked!

**Comment by gilbertsoft** [comment]
@ItaloYeltsin Thanks for your great work and improvement! I will have another look tomorrow and merge and release if everything is okey...

**Comment by gilbertsoft** [comment]
BTW If you like you can use my changes and force push again...

**Comment by ItaloYeltsin** [comment]
You're welcome! I am glad it is working! Yeap, I will squash it with the right commit right now!

File: README.md [hunk 7]
```diff
@@ -57,7 +57,10 @@ jobs:
         with:
           pattern: '^[^#].{74}'
           error: 'The maximum line length of 74 characters is exceeded.'
-          excludeDescription: '1'
+          excludeDescription: 'true' # optional: this excludes the description body of a pull request
+          excludeTitle: 'true' # optional: this excludes the title of a pull request
+          checkAllCommitMessages: 'true' # optional: this checks all commits associated with a pull request
+          accessToken: ${{ secrets.ACCESS_TOKEN }} # github access token is only required if checkAllCommitMessages is true
```
Comment (gilbertsoft) [inline README.md]:
```suggestion
          accessToken: ${{ secrets.GITHUB_TOKEN }} # github access token is only required if checkAllCommitMessages is true
```

File: __tests__/input-helper.test.ts [hunk 8]
```diff
@@ -178,27 +178,101 @@ describe('input-helper tests', () => {
...
+  })
+
+  it('should check pull_request commits', async () => {
+    mockGitHub.context = {
+      eventName: 'pull_request',
+      payload: {
+        pull_request: {
+          title: 'some-title',
+          body: 'some-body',
+          number: 62 // This pull request has exactly two commits
```
Comment (gilbertsoft) [inline __tests__/input-helper.test.ts:240]: Where is this numer `62`coming from?

**Review by gilbertsoft — CHANGES_REQUESTED** [review]
One little change with the accessToken, we should use GITHUB_TOKEN to avoid people running into the same issue we had yesterady in the first place.

File: __tests__/input-helper.test.ts (hunk 8 above)
Comment (ItaloYeltsin) [reply __tests__/input-helper.test.ts:240]: this is the pull request number [#62](https://github.com/TotalCross/totalcross/pull/62) from TotalCross. I've chosen a real case scenario to test if the commits were coming.

**Review by gilbertsoft — APPROVED** [review]
Thank you again for this great stuff!

**Pull request merged**